test-cloud/test-cloud-remote/src/test/java/org/kie/cloud/integrationtests/survival/KieServerWithWorkbenchSurvivalIntegrationTest.java

```diff
@@ -0,0 +1,160 @@
+        kieServerMgmtControllerClient.saveContainerSpec(serverInfo.getServerId(), serverInfo.getName(), CONTAINER_ID, CONTAINER_ALIAS, PROJECT_GROUP_ID, PROJECT_NAME, PROJECT_VERSION, KieContainerStatus.STARTED);
+        KieServerClientProvider.waitForContainerStart(deploymentScenario.getKieServerDeployment(), CONTAINER_ID);
+
+        logger.debug("Start process instance");
+        Long signalPid = processServicesClient.startProcess(CONTAINER_ID, SIGNALTASK_PROCESS_ID);
+        assertThat(signalPid).isNotNull().isGreaterThan(0L);
+        assertThat(queryServicesClient.findProcessInstances(0, 10)).isNotNull().hasSize(1);
+
+        scaleKieServerTo(0, serverInfo.getServerId());
+
+        logger.debug("Start process instance");
+
+        Assertions.assertThatThrownBy(() -> {
+            processServicesClient.startProcess(CONTAINER_ID, SIGNALTASK_PROCESS_ID);
+        }).isInstanceOf(KieServicesHttpException.class).hasMessageContaining("The host doesn't exist.");   
+//        try {
+//            Long noContainerPid = processServicesClient.startProcess(CONTAINER_ID, SIGNALTASK_PROCESS_ID);
+//            Assertions.fail("Container was registred on not existing server template with id " + noContainerPid);
+//        } catch (KieServerHttpRequestException ex) {
+//            // ok - requested URL does not exists
+//            System.out.println("KieServerHttpRequestException catched:\n" + ex.toString());
+//        } catch (KieServicesHttpException ex) {
+//            // ok - response code 503 Forbidden
+//            // The application is currently not serving requests at this endpoint. It may not have been started or is still starting.
+//            System.out.println("KieServicesHttpException catched:\n" + ex.toString());
+//        }
+
+        scaleKieServerTo(1, serverInfo.getServerId());
+
+        logger.debug("Wait for Kie Container start again");
+        KieServerClientProvider.waitForContainerStart(deploymentScenario.getKieServerDeployment(), CONTAINER_ID);
+
+        logger.debug("Start new process instance");
+        Long newPid = processServicesClient.startProcess(CONTAINER_ID, SIGNALTASK_PROCESS_ID);
+        assertThat(newPid).isNotNull().isGreaterThan(0L);
+
+        logger.debug("Check started processes");
+        assertThat(queryServicesClient.findProcessInstances(0, 10)).isNotNull().hasSize(2);
+
+        ProcessInstance processInstance = processServicesClient.getProcessInstance(CONTAINER_ID, signalPid);
+        assertThat(processInstance).isNotNull();
+        assertThat(processInstance.getState()).isNotNull().isEqualTo(org.kie.api.runtime.process.ProcessInstance.STATE_ACTIVE);
+    }
+
+    private void scaleKieServerTo(int count, String serverId) {
+        logger.debug("Scale Kie Server to " + count);
+        deploymentScenario.getKieServerDeployment().scale(count);
+        deploymentScenario.getKieServerDeployment().waitForScale();
+        checkServerTemplateInstanceCount(serverId, count);
```

Comment:
I would put this "checkServerTemplateInstanceCount" method directly into the code - to be more transparent about test behaviour. If such check is placed here then method name "scaleKieServerTo" has misleading name as it doesn't tell that it contains also check.